app/validators/answers/number_answer_validator.py

```diff
@@ -71,6 +74,42 @@ def validate_value_in_limits(self):
                 self.MAXIMUM_GREATER_THAN_LIMIT, value=max_value, limit=MAX_NUMBER
             )
 
+        if isinstance(answer_source_max_value, dict):
+            source = answer_source_max_value.get("source")
+            answers_with_context = self.questionnaire_schema.numeric_answer_ranges
+            value_source_max_value = (
+                self.questionnaire_schema.get_numeric_value_for_value_source(
+                    source, answer_source_max_value, answers_with_context
+                )
+            )
+            if (
+                isinstance(value_source_max_value, int)
+                and value_source_max_value < MAX_NUMBER
```

Comment:
```suggestion
                and value_source_max_value > MAX_NUMBER
```
Same here we want to add an error if the value is greater than the max limit